wire-runtime/src/main/java/com/squareup/wire/MessageSerializedForm.java

```diff
@@ -27,15 +27,16 @@
   private final Class<? extends Message> messageClass;
 
   public MessageSerializedForm(Message message, Class<? extends Message> messageClass) {
-    this.bytes = message.toByteArray();
+    this.bytes = Message.WIRE.adapter((Class<Message>) messageClass).writeBytes(message);
```

Comment:
ahhh, I think I see it. You don't know that Message and messageClass agree. Ugh.